query-connector/src/app/ui/components/header/header.tsx

```diff
@@ -186,12 +187,41 @@ const HeaderComponent: React.FC<{ authDisabled: boolean }> = ({
   );
 };
 
+export enum PAGES {
+  LANDING = "/",
+  QUERY = "/query",
+  MY_QUERIES = "/queryBuilding",
+  FHIR_SERVERS = "/fhirServers",
+  USER_MANAGEMENT = "/userManagement",
+  GROUP_MANAGEMENT = "/userManagement/userGroups",
+}
+
 const LOGGED_IN_PATHS = [
-  "/query",
-  "/queryBuilding",
-  "/fhirServers",
-  "/userManagement",
-  "/userManagement/userGroups",
+  PAGES.QUERY,
+  PAGES.MY_QUERIES,
+  PAGES.FHIR_SERVERS,
+  PAGES.USER_MANAGEMENT,
+  PAGES.GROUP_MANAGEMENT,
+];
+
+export const pagesRoleAccess: Record<string, RoleTypeValues[]> = {};
+pagesRoleAccess[PAGES.QUERY] = [
+  RoleTypeValues.SuperAdmin,
+  RoleTypeValues.Admin,
+  RoleTypeValues.Standard,
+];
+pagesRoleAccess[PAGES.MY_QUERIES] = [
+  RoleTypeValues.SuperAdmin,
+  RoleTypeValues.Admin,
+];
+pagesRoleAccess[PAGES.FHIR_SERVERS] = [
+  RoleTypeValues.SuperAdmin,
+  RoleTypeValues.Admin,
+];
+pagesRoleAccess[PAGES.USER_MANAGEMENT] = [RoleTypeValues.SuperAdmin];
+pagesRoleAccess[PAGES.GROUP_MANAGEMENT] = [
+  RoleTypeValues.SuperAdmin,
+  RoleTypeValues.Admin,
 ];
```

Comment:
@fzhao99 I am trying to get away from typing the urls directly and to centralize the pages and their access... I am wondering if all these should not even be in the `header component`.